### PR TITLE
Added details of client-side extension points with Forms 14.

### PR DESCRIPTION
**Pull request by AndyButland**
## Description

Added details of client-side extension points with Forms 14.  Specifically there are a few parts of extending Forms with custom field types etc. that now require client-side components.  I've updated the page to reference these.

## Type of suggestion

* [ ] Typo/grammar fix
* [ ] Updated outdated content
* [ ] New content
* [X] Updates related to a new version
* [ ] Other

## Product & version (if relevant)

Forms 14

## Deadline (if relevant)

Any time is fine.

### CODE REVIEW

**Comment by alina-tincas** [comment]
Hi @AndyButland thank you for the PR 🙌 

We will make sure to review it as soon as we can 🙏

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 1]
```diff
@@ -110,210 +122,375 @@ If working with Umbraco 9 or earlier versions, you'll find the `Views\Partials\F
...
+With Forms 14, aspects of the presentation and functionality of the custom field are handled by client-side components, registered via manifests:
 
 - The preview, displayed on the form definition editor.
 - The property editor UI used for editing the the submitted values via the backoffice.
+- The property editor UI used for editing settings.
+- A settings converter, that handles configuring the property editor and translating between the editor and persisted values.
+- Translations for setting labels and descriptions.
 
-These are referenced server-side using the `PreviewView` and `EditView` respectively.
+In order to create custom backoffice components for Umbraco 14, it's recommended to use a front-end build setup using Vite, TypeScript and Lit. Getting started with this is [described in the CMS documentation](https://docs.umbraco.com/umbraco-cms/tutorials/creating-your-first-extension#extension-with-vite-typescript-and-lit).
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:158]:
```suggestion
To create custom backoffice components for Umbraco 14, it's recommended to use a front-end build setup using Vite, TypeScript, and Lit. For more information, see the [Creating your first extension](https://docs.umbraco.com/umbraco-cms/tutorials/creating-your-first-extension#extension-with-vite-typescript-and-lit) article.
```

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 2]
```diff
@@ -110,210 +122,375 @@ If working with Umbraco 9 or earlier versions, you'll find the `Views\Partials\F
...
 
-For the edit view, you can use a built-in property editor UI, one from a package, or a custom one registered with your solution.
+### Field preview
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:160]:
```suggestion
### Field Preview
```

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 3]
```diff
@@ -110,210 +122,375 @@ If working with Umbraco 9 or earlier versions, you'll find the `Views\Partials\F
...
+    meta: {
+        label: 'Number Input',
+        icon: 'icon-autofill',
+    },
+};
+export const manifests = [numberPropertyEditorManifest];
 ```
 
-The area aliases for the other provider types are as follows:
+### Setting value editor
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:302]:
```suggestion
### Setting Value Editor
```

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 4]
```diff
@@ -110,210 +122,375 @@ If working with Umbraco 9 or earlier versions, you'll find the `Views\Partials\F
...
+        label: 'Color Input',
+        icon: 'icon-autofill',
+    },
+};
+
+export const manifests = [colorPropertyEditorManifest];
+```
 
-You may want to consider registering a settings value converter. This is a client-side component that is registered in a manifest. It converts between the setting value required for the editor and the value persisted with the form definition. A converter defines three methods:
+### Setting value converter
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:376]:
```suggestion
### Setting Value Converter
```

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 5]
```diff
@@ -110,210 +122,375 @@ If working with Umbraco 9 or earlier versions, you'll find the `Views\Partials\F
...
+  name: "Slider Value Converter",
+  propertyEditorUiAlias: "My.PropertyEditorUi.Slider",
+  api: SliderSettingValueConverter,
+};
+
+export const manifests = [sliderValueConverterManifest];
 ```
 
-`SupportsPlaceholders` is a flag indicating whether the setting can contain ["magic string" placeholders](../magic-strings.md) and controls whether they are parsed on rendering.
+### Language files
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:421]:
```suggestion
### Language Files
```

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 6]
```diff
@@ -110,210 +122,375 @@ If working with Umbraco 9 or earlier versions, you'll find the `Views\Partials\F
...
+    meta: {
+      culture: "en-us",
+    },
+    js: () => import("./en-us.js"),
+  },
+];
+export const manifests = [...localizationManifests];
 ```
+
+### Registering the components
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:476]:
```suggestion
### Registering the Components
```

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 7]
```diff
@@ -110,210 +122,375 @@ If working with Umbraco 9 or earlier versions, you'll find the `Views\Partials\F
...
+
+Finally you will need an entry point to your client-side components that will register the manifests with Umbraco's extension registry. For example:
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:478]:
```suggestion
Finally, you will need an entry point to your client-side components that will register the manifests with Umbraco's extension registry. For example:
```

**Review by eshanrnh — APPROVED** [review]
Looks great, @AndyButland 🙌 Few minor suggestions which I'll incorporate and merge it in.

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 8]
```diff
@@ -76,32 +80,43 @@ using Umbraco.Cms.Core.Composing;
...
-            builder.WithCollectionBuilder<FieldCollectionBuilder>()
-                .Add<MyCustomField>();
-        }
+        builder.WithCollectionBuilder<FieldCollectionBuilder>()
+            .Add<SliderFieldType>();
     }
 }
 ```
 
 ## Partial view
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:95]:
```suggestion
## Partial View
```

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 9]
```diff
@@ -110,210 +125,375 @@ If working with Umbraco 9 or earlier versions, you'll find the `Views\Partials\F
...
+
+`HtmlEncodeReplacedPlaceholderValues` takes effect only if `SupportsPlaceholders` is `true`. It controls whether the replaced placeholder values should be HTML encoded (as is necessary for rendering within content from a rich text editor).
+
+`IsMandatory` if set to `true` will provide client-side validation in the backoffice to ensure the value is completed.
+
+When creating a field or other provider type, you might choose to inherit from an existing class. This could be if one of the types provided with Umbraco Forms almost meets your needs but you want to make some changes.
+
+All setting properties for the Forms provider types are marked as `virtual`, so you can override them and change the setting values:
+
 ## Umbraco backoffice components
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:151]:
```suggestion
## Umbraco Backoffice Components
```

File: 14/umbraco-forms/developer/extending/adding-a-fieldtype.md [hunk 10]
```diff
@@ -110,210 +125,375 @@ If working with Umbraco 9 or earlier versions, you'll find the `Views\Partials\F
...
+    name: "Forms UUI Slider Field Preview",
+    api: MyFieldPreviewSliderElement,
+    element: () => import('./slider-preview.element.js')
   };
+
+  export const manifests = [sliderPreviewManifest];
 ```
 
-## Field settings
+### Field editor
```
Comment (eshanrnh) [inline 14/umbraco-forms/developer/extending/adding-a-fieldtype.md:239]:
```suggestion
### Field Editor
```

**Pull request merged**